### PR TITLE
Generate HTML or Text coverage for the latest version

### PR DESCRIPTION
**Pull request by walkero-gr**
This Pull Request is based on the PHPUnit coverage as I introduced it at #94 , but now it is for the latest version of Better-PHPUnit.

- Added a new configuration variable, where the user can select the format of the Coverage report, which by default is text.
- Reduced the commands to two than four, so that it is easier for the users
- Added an Error Message for the user in case the coverage path is not set when the user runs a test with coverage. The test is not prevented from running.

hope this implementation will be useful for others too.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,6 +90,23 @@
...
                 },
+                "better-phpunit.coveragePath": {
+                    "type": [
+                        "string",
+                        "null"
+                    ],
+                    "default": null,
+                    "description": "Relative path for the generated coverage files, based on the project folder. Ex: 'test/coverage'"
+                },
+				"better-phpunit.coverageFormat": {
```
Comment (calebporzio) [inline package.json]: can you fix the weird spacing here please

File: src/extension.js [hunk 2]
```diff
@@ -77,6 +85,24 @@ function setGlobalCommandInstance(commandInstance) {
...
+async function runWithCoverage(options) {
+    let command;
+
+    if (vscode.workspace.getConfiguration("better-phpunit").get("ssh.enable")) {
+        command = new RemotePhpUnitCommand( options );
+    } else if (vscode.workspace.getConfiguration("better-phpunit").get("docker.enable")) {
+        command = new DockerPhpUnitCommand( options );
+    } else {
+        command = new PhpUnitCommand( options );
+    }
```
Comment (calebporzio) [inline src/extension.js]: This "runWithCoverage" method seems redundant - maybe there is a better abstraction that doesn't duplicate this entire conditional block.

File: src/phpunit-command.js [hunk 3]
```diff
@@ -20,8 +24,8 @@ module.exports = class PhpUnitCommand {
         suiteSuffix = suiteSuffix ? ' '.concat(suiteSuffix) : '';
 
         this.lastOutput = this.runFullSuite
-            ? `${this.binary}${suiteSuffix}${this.suffix}`
-            : `${this.binary} ${this.file}${this.filter}${this.configuration}${this.suffix}`;
+            ? `${this.binary}${suiteSuffix}${this.suffix}${this.coverage}`
```
Comment (calebporzio) [inline src/phpunit-command.js]: thinking "this.suffix" should probably go after "this.coverage"?

**Comment by calebporzio** [comment]
Hey @walkero-gr - thanks for contributing! Left some notes on the PR for you if you're interested in cleaning it up a bit.

File: package.json (hunk 1 above)
Comment (walkero-gr) [reply package.json]: Fixed at my last commit. Maybe it would be useful to introduce a .editorconfig file.

File: src/phpunit-command.js (hunk 3 above)
Comment (walkero-gr) [reply src/phpunit-command.js]: Changed at my last commit

File: src/extension.js (hunk 2 above)
Comment (walkero-gr) [reply src/extension.js]: I am not sure how you would like this to be. If I would use something like `vscode.commands.executeCommand`, to execute the default `run` and `run-suite` commands, from what I understand I must change the way they set the options and to get parameters as well. Unless there is an other way to alter the `command` variable.

**Comment by walkero-gr** [comment]
@calebporzio Hello there. Would you like to provide some more info on the above, so that we can move this pull request forward?
Thank you for your time.

**Pull request closed without merging**